landing page setup

simonious1989 opened the pull request.

simonious1989 commented:
@flumonion I have created one landing page template which may require additional styling.
The example is two columns, text left and form on the right.

flumonion commented:
Will make some suggestions/recommendations and commit them to this branch.

flumonion commented:
Made some changes.
- shortened url
- made page type descriptive
- changed styling in line with other pages

We should:-
- Create partials for shared code (e.g. artticles and video) this will make it easier to add to other templates in future.
- add name tag to netlify forms.
- ensure we are not duplicate already created form partials.
- remove the word "landing" from the form partial, its just a form and can be included anywhere on site, not specifically for landing pages.

flumonion commented:
@simonious1989 forgot to also include a way to add a "do not index" for crawlers on these pages only...

simonious1989 commented:
@flumonion 

> Create partials for shared code (e.g. artticles and video) this will make it easier to add to other templates in future.
I have created two new partials:
 - video
 - latest-articles 

> add name tag to netlify forms.
I have updated the form to include a form name. 
Should I do this on the other form partials for consistency?

> ensure we are not duplicate already created form partials.
I did create a new form partial and remove the "text" input.

> remove the word "landing" from the form partial, its just a form and can be included anywhere on site, not specifically for landing pages.
I have done this and renamed it to "quick-contact"

> forgot to also include a way to add a "do not index" for crawlers on these pages only...
I looked into that last night and came across the [following approach](https://github.com/dereckcurry/hugo-robots-meta-tags).
I am not sure if this means we need to go and update all pages now with the suggested front matter.

Also if these pages are not being indexed, we can remove the SEO front matter.

flumonion commented:
> Should I do this on the other form partials for consistency?

Yes

simonious1989 commented:
@flumonion 

I have added a check in the in the head.html for the `robotsNoFollow` param.
have done a check locally and all other pages without this param are not affected, meaning they still retain:

`<META NAME="ROBOTS" CONTENT="INDEX, FOLLOW">`

All forms now have name tags.
Please could you review and merge.

flumonion commented on themes/s2s/layouts/partials/latest-articles.html:
```diff
@@ -0,0 +1,33 @@
+<section class="section section--gray section--size4">
```
In the partial I don't think you should have the wrapper html.
It should only include the cards div onwards.
This means you can list articles in different ways.
i.e. you could have them run across full screen, or smaller etc.

If we wanted to get fancy, we could include [variables](https://gohugo.io/templates/partials/#variable-scoping) in the partials that can be set outside the partial by the page adding the partial. In this example the partial would be in charge of listing articles, the page adding the partial could decide what articles to add.



For example, on one page we could listt

simonious1989 replied:
@flumonion I have updated this partial and renamed to `article-card.html`
I can refactor everywhere else where this article section is being used on the site now also.

flumonion commented on themes/s2s/layouts/partials/quick-contact.html:
```diff
@@ -0,0 +1,45 @@
+<div class="container">
+    <div class="form form--primary form--primary-alt">
+        <form method="post" name="quick-contact" data-netlify="true">
```
partial should not have "container" class.

simonious1989 replied:
@flumonion I have removed the "container" class from the partial.

simonious1989 commented:
@flumonion 

I have made the following changes:

- created a `call.html` partial 
- updated `quick-contact` partial to remove container class
- used .Scratch to get and set the form name in the lp-form template. (I can change this to rather use a variable so it can be set in the page from matter).
- updated the `article.html` partial 
- updated the contact links on the footer
- updated the lp full form copy
